src/import.c

```diff
@@ -0,0 +1,104 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "storage.h"
+#include "util.h"
+#include "module.h"
+#include "import.h"
+#include "entry.h"
+#include <string.h>
+#include <stdio.h>
+
+
+/**
+ * CacheImport:
+ * @shadow_path: Path to shadow file.
+ * @shadow_hash: Hashed password. 
+ * @storage_path: Path to store policy cache entry.
+ * @username: Username to create cache entry for.
+ * @error: Errors.
+ *
+ * Returns: #True if CacheEntry was created or #False on failure.
+ */
+
+gboolean CacheImport(const char *shadow_path, const char *storage_path, const char *username, GError **error) {
+  if (!shadow_path) {
+    shadow_path = "/etc/shadow";
+  }
+  
+  if (!storage_path) {
+    storage_path = DEFAULT_STORAGE_PATH;
+  }
+
+  GError *errors = NULL;
+  // Get user's hashed password.
+  GBytes* shadow_hash = CacheUtilReadShadowFile(shadow_path, username, &errors);
+
+  // Create a new Cache Entry.
+  CacheEntry *entry = CacheEntryNew();
+
+  // Set the newly created entry hash using the /etc/shadow hash.
+  CacheEntryHashSet(entry, CACHE_ENTRY_ALGORITHM_CRYPT, shadow_hash);
+
+  GError *tmp_error = NULL;
+  CacheStorage *storage = CacheStorageNew(storage_path);
+  if (!CacheStoragePutEntry(storage, username, entry, &tmp_error)) {
+    // TODO: Log error message
+    g_error_free(tmp_error);
+    g_set_error(error, CACHE_IMPORT_ERROR, CACHE_IMPORT_ERROR_STORAGE_FAIL, "Could not store cache entry.");
+    return FALSE;
+  }
+  return TRUE;
+}
+
+#ifndef CACHE_IMPORT_TESTING
+int main(int argc, char *argv[]) {
+  int exit_code = 0;
+  GError *error = NULL;
+  GOptionContext *context = NULL;
+ 
+  static gchar* shadow_path = NULL;
+  static gchar* storage_path = NULL;
+  static GOptionEntry entries[] = 
+    {
+        { "shadow-path", 'p', 0, G_OPTION_ARG_STRING, &shadow_path, "Path to search for hashed passwords."},
+        { "storage-path", 's', 0, G_OPTION_ARG_STRING, &storage_path, "Path to storage policy cache."},
+	{ NULL }
+    };
+
+  context = g_option_context_new("- import shadow hash module");
+  g_option_context_add_main_entries(context, entries, NULL);
+  if (!g_option_context_parse(context, &argc, &argv, &error)) {
+    goto done;
+  }
+    
+  char *user = argv[1];
+  if (user != NULL) { 
+    CacheImport(shadow_path, storage_path, user, &error);    
+  }
+      
+  done:
+    if (error) {
+      g_printerr("option parsing failed %s\n", error->message);
```

Comment:
CacheImport can set error too, but the message only mentions option parsing.

Also, add a colon right before the error message.